routers/api/packages/rubygems/rubygems.go

```diff
@@ -217,7 +218,11 @@ func UploadPackageFile(ctx *context.Context) {
 
 	rp, err := rubygems_module.ParsePackageMetaData(buf)
 	if err != nil {
-		apiError(ctx, http.StatusInternalServerError, err)
+		if errors.Is(err, util.ErrInvalidArgument) {
+			apiError(ctx, http.StatusBadRequest, err)
+		} else {
+			apiError(ctx, http.StatusInternalServerError, err)
+		}
```

Comment:
I've seen this block now often enough.
What about an error handling function that takes `ctx` and `err` as parameters?

Reply:
`apiError` is always different and must then be passed as parameter too.

Reply:
Ah, makes sense, even though I wouldn't have expected it as it always _looks_ the same.